src/Subscription/SubscriptionHandler.php

```diff
@@ -3,23 +3,52 @@
 namespace AppBundle\Subscription;
 
 use AppBundle\Entity\Adherent;
+use AppBundle\Entity\SubscriptionType;
+use AppBundle\Repository\AdherentRepository;
 use AppBundle\Repository\SubscriptionTypeRepository;
 
 class SubscriptionHandler
 {
-    private $repository;
+    public const ACTION_TYPE_SUBSCRIBE = 'subscribe';
+    public const ACTION_TYPE_UNSUBSCRIBE = 'unsubscribe';
 
-    public function __construct(SubscriptionTypeRepository $repository)
+    private $adherentRepository;
+    private $subscriptionTypeRepository;
+
+    public function __construct(AdherentRepository $adherentRepository, SubscriptionTypeRepository $subscriptionTypeRepository)
     {
-        $this->repository = $repository;
+        $this->adherentRepository = $adherentRepository;
+        $this->subscriptionTypeRepository = $subscriptionTypeRepository;
+    }
+
+    public function changeSubscription(string $type, string $email, string $listId): void
+    {
+        /** @var Adherent $adherent */
+        $adherent = $this->adherentRepository->findOneByEmail($email);
+        if (!$adherent) {
+            throw new \RuntimeException(sprintf('There is no adherent with email address "%s".', $email));
+        }
+
+        /** @var SubscriptionType $subscriptionType */
+        $subscriptionType = $this->subscriptionTypeRepository->findOneBy(['externalId' => $listId]);
+        if (!$subscriptionType) {
+            throw new \RuntimeException(sprintf('There is no subscription type with external service id "%s".', $listId));
+        }
+
+        $hasSubscription = in_array($subscriptionType->getCode(), $adherent->getEmailsSubscriptions());
```

Comment:
Wouldn't it be better to use an adherent method for it? We might need this logic again elsewhere I guess. What about:
```php
$hasSubscription = $adherent->hasSubscriptionType($type);
```

Reply:
Done.